std/algorithm/sorting.d

```diff
@@ -1048,147 +1048,277 @@ unittest
-        assert(!empty);
-        foreach (i, U; Rs)
+        private size_t backIndex()
         {
-            if (i < _crt) continue;
-            // found _crt
-            assert(!_r[i].empty);
-            _r[i].popFront();
-            adjustPosition();
-            return;
+            import std.functional : binaryFun;
+            size_t bestIndex = size_t.max; // indicate undefined
+            Unqual!ElementType bestElement;
+            foreach (i, _; Rs)
+            {
+                if (!source[i].empty)
+                {
+                    if (bestIndex == size_t.max || // either this is the first or
+                        binaryFun!less(bestElement, source[i].back))
+                    {
+                        bestIndex = i;
+                        bestElement = source[i].back;
+                    }
+                }
+            }
+            return bestIndex;
         }
-        assert(false);
-    }
 
-    ///
-    @property auto ref ElementType front()
-    {
-        assert(!empty);
-        // Assume _crt is correct
-        foreach (i, U; Rs)
+        void popBack()
         {
-            if (i < _crt) continue;
-            assert(!_r[i].empty);
-            return _r[i].front;
+            final switch (_lastBackIndex)
+            {
+                foreach (i, _; Rs)
+                {
+                case i:
+                    source[i].popBack();
+                    break;
+                }
+            }
+            _lastBackIndex = backIndex;
         }
-        assert(false);
     }
 
-    static if (allSatisfy!(isForwardRange, Rs))
+    static if (allSatisfy!(isForwardRange, staticMap!(Unqual, Rs)))
     {
-        ///
         @property auto save()
         {
-            auto ret = this;
-            foreach (ti, elem; _r)
+            auto result = this;
+            foreach (i, _; Rs)
             {
-                ret._r[ti] = elem.save;
+                result.source[i] = result.source[i].save;
             }
-            return ret;
+            return result;
         }
     }
 
     static if (allSatisfy!(hasLength, Rs))
     {
-        ///
         @property size_t length()
         {
             size_t result;
-            foreach (i, U; Rs)
+            foreach (i, _; Rs)
             {
-                result += _r[i].length;
+                result += source[i].length;
             }
             return result;
         }
 
-        ///
         alias opDollar = length;
     }
+
+    public Rs source;
+    private size_t _lastFrontIndex = size_t.max;
+    static if (isBidirectional)
+    {
+        private size_t _lastBackIndex = size_t.max;
+    }
 }
 
-/// Ditto
-Merge!(less, Rs) merge(alias less = "a < b", Rs...)
-(Rs rs)
+/**
+   Merge multiple sorted ranges $(D rs) with less-than predicate function $(D
+   pred) into one single sorted output range containing the sorted union of the
+   elements of inputs. Elements in the output are not unique, that is the length
```

Comment:
s/that is/consequently/

Better yet, rephrase the sentence: "Duplicates are not eliminated, meaning that the total number of elements in the result of `merge` is the sum of all elements in the ranges passed to it."